### PR TITLE
Port BusBypass.scala to Chisel 3

### PR DESCRIPTION
**Pull request by sinofp**
**Related issue**: #3083 

**Type of change**: other enhancement

**Impact**: no functional change

**Development Phase**: implementation

**Release Notes**

### CODE REVIEW

File: src/main/scala/devices/tilelink/BusBypass.scala [hunk 1]
```diff
@@ -92,7 +92,7 @@ class TLBusBypassBar(dFn: TLSlavePortParameters => TLSlavePortParameters)(implic
     out0.d.ready := in.d.ready &&  bypass
     out1.d.ready := in.d.ready && !bypass
     in.d.valid   := Mux(bypass, out0.d.valid, out1.d.valid)
-    def cast(x: TLBundleD) = { val out = Wire(in.d.bits); out <> x; out }
+    def cast(x: TLBundleD) = { val out = WireDefault(in.d.bits); out <> x; out }
```
Comment (sequencer) [inline src/main/scala/devices/tilelink/BusBypass.scala:95]: I don't think this is OK.
`Wire(in.d.bits)` only use the hardware type and generate a `Wire` hardware, but what you are proposing here is giving `out` a non constant value with `in.d.bits`.

**Review by sequencer — CHANGES_REQUESTED** [review]
need fix the `cast` function.

File: src/main/scala/devices/tilelink/BusBypass.scala (hunk 1 above)
Comment (sinofp) [reply src/main/scala/devices/tilelink/BusBypass.scala:95]: Chisel.Wire is chisel3.WireDefault:
https://github.com/chipsalliance/chisel3/blob/fc970ca28e562f2ea3ba160963604ea3deaf3467/src/main/scala/chisel3/compatibility.scala#L72-L80

Comment (sequencer) [reply src/main/scala/devices/tilelink/BusBypass.scala:95]: oh fk

**Review by sequencer — APPROVED** [review]

**Comment by sequencer** [comment]
> Chisel.Wire is chisel3.WireDefault

Sure........

**Pull request merged**